src/redprl/machine.fun

```diff
@@ -557,7 +557,103 @@ struct
+                         fun projFromOne s = Syn.intoCoe {dir = (Syn.intoDim1, s), ty = (v, b), coercee = coercee}
+                         fun fiberFromOne s = Syn.intoFst @@ Syn.intoApp (Syn.intoSnd (substVar (s, v) e), projFromOne s)
+                         fun nFromOne s t =
+                           let
+                             val w = Sym.named "w"
+                           in
+                             Syn.intoHcom (* t is the dimension to fix the end. *)
+                               {dir = (Syn.intoDim1, t),
+                                ty = substVar (s, v) b,
+                                cap = projFromOne s,
+                                tubes =
+                                  [ ((s, Syn.intoDim0), (w, Syn.into @@ Syn.PATH_APP (Syn.intoSnd (fiberFromOne s), VarKit.toDim w)))
+                                  , ((s, Syn.intoDim1), (w, projFromOne s)) ]}
+                           end
+                       in
+                         branchOnDim stability syms' (#1 dir)
+                           (Syn.into @@ Syn.VIN (#2 dir, coercee, nFromZero (#2 dir)))
+                           (Syn.into @@ Syn.VIN (#2 dir, Syn.intoFst (fiberFromOne (#2 dir)), nFromOne (#2 dir) Syn.intoDim1))
+                           (fn y =>
+                              let
+                                fun base x =
+                                  let
+                                    val w = Sym.named "w"
+                                  in
+                                    Syn.intoCom
+                                      {dir = (#1 dir, x), ty = (v, b),
+                                       cap = Syn.into @@ Syn.VPROJ (#1 dir, coercee, Syn.intoFst @@ substVar (#1 dir, v) e),
+                                       tubes =
+                                         [ ((#1 dir, Syn.intoDim0), (w, nFromZero (VarKit.toDim w)))
+                                         , ((#1 dir, Syn.intoDim0), (w, projFromOne (VarKit.toDim w))) ]}
+                                  end
+                                val wallZero =
+                                  let
+                                    val z = Sym.named "z"
+                                    val m = Syn.intoCoe
+                                      {dir = (Syn.intoDim0, VarKit.toDim y),
+                                       ty = (y, substVar (Syn.intoDim0, v) a),
+                                       coercee = substVar (Syn.intoDim0, y) coercee}
+                                  in
+                                    Syn.intoAnonTuple
+                                      [ m
+                                      , Syn.into @@ Syn.PATH_ABS (z,
+                                          Syn.intoCom
+                                            {dir = (Syn.intoDim0, VarKit.toDim y), (* should I worry about name collision here? *)
+                                             ty = (y, substVar (Syn.intoDim0, v) b),
```

Comment:
@jonsterling Should I worry about the `y` in `dir` colliding with the bound variable in `ty`?

Reply:
It should be safe as-is.

Reply:
Do you mean that, after `Syn.into`, any foreseeable usage should see the `y` in `ty` as a refreshed (bound) variable?

Reply:
Yeah, `Syn.into` will call `Abt.into`, which turns `y` into a de bruijn index; so, in the type, this bound `y` is no longer the same as the `y` that is mentioned in `dir`.